File: crime-analytics-backend/src/main/java/com/crimeanalytics/service/CriminalService.java
package com.crimeanalytics.service;

import com.crimeanalytics.dto.request.CriminalRequestDTO;
import com.crimeanalytics.dto.response.CriminalResponseDTO;

import java.util.List;

public interface CriminalService {

    CriminalResponseDTO createCriminal(CriminalRequestDTO requestDTO);

    List<CriminalResponseDTO> getAllCriminals();

    CriminalResponseDTO getCriminalById(Long id);

    CriminalResponseDTO updateCriminal(
            Long id,
            CriminalRequestDTO requestDTO
    );

    void deleteCriminal(Long id);
}

File: crime-analytics-frontend/src/services/criminalService.js
import axiosInstance from '../api/axiosInstance';

export const criminalService = {
  async getAllCriminals() {
    const response = await axiosInstance.get('/criminals');
    return response.data;
  },

  async getCriminalById(id) {
    const response = await axiosInstance.get(`/criminals/${id}`);
    return response.data;
  },

  async createCriminal(data) {
    // data: { name, alias, gender, age, address, phoneNumber, criminalStatus, description }
    const response = await axiosInstance.post('/criminals', data);
    return response.data;
  },

  async updateCriminal(id, data) {
    const response = await axiosInstance.put(`/criminals/${id}`, data);
    return response.data;
  },

  async deleteCriminal(id) {
    const response = await axiosInstance.delete(`/criminals/${id}`);
    return response.data;
  }
};
